### PR TITLE
docs: KFServing and Seldon Pipeline Examples with Explainers, Outlier Detectors and Drift Detectors

### PR DESCRIPTION
**Pull request by cliveseldon**
**Description of your changes:**
Adds contrib end-to-end pipelines for KFServing and Seldon that show
 * Image and tabular models
    * CIFAR10 and UCI Census dataset
 * Train and test model explainers
 * Train and test outlier detectors
 * Train and test drift detectors

### CODE REVIEW

File: samples/contrib/e2e-outlier-drift-explainer/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# End to end Deployments with Drift, Outlier and Explainers
```
Comment (animeshsingh) [inline samples/contrib/e2e-outlier-drift-explainer/README.md:1]:
```suggestion
# Build, Train and Deploy with Drift, Outlier Detectors and Explainers enabled on KFServing and Seldon
```

File: samples/contrib/e2e-outlier-drift-explainer/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+gcloud beta filestore instances create ${FS}     --project=${PROJECT}     --zone=${ZONE}     --tier=STANDARD     --file-share=name="volumes",capacity=1TB     --network=name="default",reserved-ip-range="10.0.0.0/29"
+
+FSADDR=$(gcloud beta filestore instances describe ${FS} --project=${PROJECT} --zone=${ZONE} --format="value(networks.ipAddresses[0])")
+
+helm install nfs-cp stable/nfs-client-provisioner --set nfs.server=${FSADDR} --set nfs.path=/volumes --namespace=kubeflow 
+
+kubectl rollout status  deploy/nfs-cp-nfs-client-provisioner -n kubeflow
+```
+
+If you build the pipeline python using Kale from the notebook you will at present need to modify the created pyhton and change the Kale `VolumeOp` by adding a `storage_class` for the NFS PV, for example:
```
Comment (animeshsingh) [inline samples/contrib/e2e-outlier-drift-explainer/README.md:29]:
```suggestion
If you build the pipeline Python DSL using Kale from the notebook you will at present need to modify the created pyhton and change the Kale `VolumeOp` by adding a `storage_class` for the NFS PV, for example:
```

File: samples/contrib/e2e-outlier-drift-explainer/kfserving/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# KFServing End to End Deployment Examples
+
+ * [CIFAR10 Image Classifier with Explainer, Outlier detector and drift detector](#cifar10-image-classification-model)
```
Comment (animeshsingh) [inline samples/contrib/e2e-outlier-drift-explainer/kfserving/README.md:3]:
```suggestion
 * [CIFAR10 Image Classifier with Explainer, Outlier Detector and Drift Detector](#cifar10-image-classification-model)
```

File: samples/contrib/e2e-outlier-drift-explainer/kfserving/README.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    * [Kale annotated Jupyter Notebook](./seldon_e2e_cifar10.ipynb)
+    * [Pipeline](./seldon_e2e_cifar10.kale.py)
+      * Assumes `storage_class="nfs-client"`
+
+![pipeline](cifar10-pipeline.png)
+
+
+### Model Deployment and Predictions
+
+Deploy a trained tensorflow model and call the Seldon API to get predictions.
```
Comment (animeshsingh) [inline samples/contrib/e2e-outlier-drift-explainer/kfserving/README.md:27]:
```suggestion
Deploy a trained Tensorflow model and call the Seldon API to get predictions.
```

File: samples/contrib/e2e-outlier-drift-explainer/seldon/README.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    * [Kale annotated Jupyter Notebook](./seldon_e2e_cifar10.ipynb)
+    * [Pipeline](./seldon_e2e_cifar10.kale.py)
+      * Assumes `storage_class="nfs-client"`
+
+![pipeline](cifar10-pipeline.png)
+
+
+### Model Deployment and Predictions
+
+Deploy a trained tensorflow model and call the Seldon API to get predictions.
```
Comment (animeshsingh) [inline samples/contrib/e2e-outlier-drift-explainer/seldon/README.md:27]:
```suggestion
Deploy a trained Tensorflow model and call the Seldon API to get predictions.
```

Comment (animeshsingh) [reply samples/contrib/e2e-outlier-drift-explainer/seldon/README.md:27]: Can this TF model be used with KFServing as well?

Comment (cliveseldon) [reply samples/contrib/e2e-outlier-drift-explainer/seldon/README.md]: Yes, this was a mistake I forgot to add files. Thanks for spotting!

**Comment by cliveseldon** [comment]
> Thanks @cliveseldon - this looks great. Can we remove the outputs from the notebook before checking in?

Done.

**Comment by animeshsingh** [comment]
thanks - one more thing I do want to highlight is that Pipeline samples are more catered to use the "Components" as a unit to be used, for e.g. we have created a "component" for KFServing
https://github.com/kubeflow/pipelines/blob/master/components/kubeflow/kfserving/component.yaml

Breaking notebook cells into different pipelines steps mostly doesn't give a prescriptive view for the folks who are writing the pipeline DSLs .

Would be good to revisit this after this initial merge to see how we can leverage that as a unit and gateway to interact with KFServing. 

/lgtm

**Comment by animeshsingh** [comment]
@Ark-kun @Bobgy @rmgogogo please chime in

**Comment by Bobgy** [comment]
Sure

/approve
/lgtm

**Pull request merged**